### PR TITLE
Improve test coverage of core unit tests

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Add missing UT on some core module classes, and improve some UT for maintainability

### CODE REVIEW

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/utils/MiscUtils.kt [hunk 1]
```diff
@@ -51,7 +51,7 @@ internal inline fun retryWithDelay(
                     { "Internal I/O operation failed" },
                     e
                 )
-                return false
+                false
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/utils/MiscUtils.kt:54]: such a tiny typo, but such a big functional change

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ okHttp = "4.12.0"
 kronosNTP = "0.0.1-alpha11"
 
 # Android
-androidToolsPlugin = "8.6.1"
+androidToolsPlugin = "8.5.2"
```
Comment (0xnm) [inline gradle/libs.versions.toml:10]: why are we doing downgrade?

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/ThreadExtTest.kt [hunk 3]
```diff
@@ -23,7 +24,7 @@ import org.mockito.junit.jupiter.MockitoExtension
 @ForgeConfiguration(Configurator::class)
 internal class ThreadExtTest {
 
-    @Test
+    @RepeatedTest(16)
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/ThreadExtTest.kt:27]: is this change on purpose or some debugging leftover?

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/MiscUtilsTest.kt [hunk 4]
```diff
@@ -90,40 +97,58 @@ internal class MiscUtilsTest {
...
+
+    @Test
+    fun `M repeat until success W retryWithDelay { exception }`(
+        @Forgery exception: Exception,
+        forge: Forge
+    ) {
+        // Given
+        val fakeDelay = TimeUnit.SECONDS.toNanos(forge.aLong(min = 0, max = 2))
+        val mockedBlock: () -> Boolean = mock()
+        whenever(mockedBlock.invoke()).thenAnswer { throw exception }.thenReturn(true)
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/MiscUtilsTest.kt:134]: probably can be shorter a bit

```suggestion
        whenever(mockedBlock.invoke()).thenThrow(exception).thenReturn(true)
```

**Review by 0xnm — APPROVED** [review]

**Review by jonathanmos — APPROVED** [review]

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/MiscUtilsTest.kt (hunk 4 above)
Comment (xgouchet) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/utils/MiscUtilsTest.kt:134]: Indeed 👍

**Review by 0xnm — APPROVED** [review]
